satpy/tests/reader_tests/test_mersi2_l1b.py

```diff
@@ -513,31 +531,195 @@ def test_fy3d_250_resolutions(self):
             ds_id = make_dataid(name=band_name, resolution=250)
             res = get_key(ds_id, available_datasets,
                           num_results=num_results, best=False)
-            self.assertEqual(num_results, len(res))
+            assert num_results == len(res)
             ds_id = make_dataid(name=band_name, resolution=1000)
             with pytest.raises(KeyError):
-                res = get_key(ds_id, available_datasets,
-                              num_results=num_results, best=False)
+                get_key(ds_id, available_datasets, num_results=num_results, best=False)
 
         res = reader.load(['1', '2', '3', '4', '5', '20', '24', '25'])
-        self.assertEqual(6, len(res))
-        self.assertRaises(KeyError, res.__getitem__, '5')
-        self.assertRaises(KeyError, res.__getitem__, '20')
-        self.assertEqual((2 * 40, 2048 * 2), res['1'].shape)
-        self.assertEqual('reflectance', res['1'].attrs['calibration'])
-        self.assertEqual('%', res['1'].attrs['units'])
-        self.assertEqual((2 * 40, 2048 * 2), res['2'].shape)
-        self.assertEqual('reflectance', res['2'].attrs['calibration'])
-        self.assertEqual('%', res['2'].attrs['units'])
-        self.assertEqual((2 * 40, 2048 * 2), res['3'].shape)
-        self.assertEqual('reflectance', res['3'].attrs['calibration'])
-        self.assertEqual('%', res['3'].attrs['units'])
-        self.assertEqual((2 * 40, 2048 * 2), res['4'].shape)
-        self.assertEqual('reflectance', res['4'].attrs['calibration'])
-        self.assertEqual('%', res['4'].attrs['units'])
-        self.assertEqual((2 * 40, 2048 * 2), res['24'].shape)
-        self.assertEqual('brightness_temperature', res['24'].attrs['calibration'])
-        self.assertEqual('K', res['24'].attrs['units'])
-        self.assertEqual((2 * 40, 2048 * 2), res['25'].shape)
-        self.assertEqual('brightness_temperature', res['25'].attrs['calibration'])
-        self.assertEqual('K', res['25'].attrs['units'])
+        assert len(res) == 6
+        with pytest.raises(KeyError):
+            res.__getitem__('5')
+        with pytest.raises(KeyError):
+            res.__getitem__('20')
+        _test_helper(res)
+        assert res['24'].shape == (2 * 40, 2048 * 2)
+        assert res['24'].attrs['calibration'] == 'brightness_temperature'
+        assert res['24'].attrs['units'] == 'K'
+        assert res['25'].shape == (2 * 40, 2048 * 2)
+        assert res['25'].attrs['calibration'] == 'brightness_temperature'
+        assert res['25'].attrs['units'] == 'K'
+
+
+class TestFY3EMERSI2L1B(MERSI2L1BTester):
```

Comment:
Thoughts on adding `LL` to the name? Too hard to read? I just wanted to make it extra clear this is for a similar yet completely different instrument to the other MERSI2 instrument.

```suggestion
class TestFY3EMERSI2LLL1B(MERSI2L1BTester):
```

Reply:
This is a mistake, it should be `TestFY3EMERSILLL1B`. Which is also fairly incomprehensible. Can change that if you have a better suggestion.

Reply:
Ah with the clarification that the instrument is MERSI-LL and not MERSI-2-LL, maybe just remove the `2` from the existing name?